common/utils.cpp

```diff
@@ -121,52 +123,65 @@ bool guessResolution(const char* filename, int& w, int& h)
     return w && h;
 }
 
+struct RoslutionEntry {
+    uint32_t fourcc;
+    uint32_t planes;
+    //multiple to half width
+    //if it equals 1, you need divide width with 2
+    //if it equals 4, you need wultiple width with 2
+    uint32_t widthMultiple[3];
+    uint32_t heightMultiple[3];
+};
+
+const static RoslutionEntry resolutionEntrys[] = {
+    { VA_FOURCC_I420, 3, { 2, 1, 1 }, { 2, 1, 1 } },
+    { VA_FOURCC_YV12, 3, { 2, 1, 1 }, { 2, 1, 1 } },
+    { VA_FOURCC_IMC3, 3, { 2, 1, 1 }, { 2, 1, 1 } },
+    { VA_FOURCC_422H, 3, { 2, 1, 1 }, { 2, 2, 2 } },
+    { VA_FOURCC_422V, 3, { 2, 2, 2 }, { 2, 1, 1 } },
+    { VA_FOURCC_444P, 3, { 2, 2, 2 }, { 2, 2, 2 } },
+    { VA_FOURCC_YUY2, 1, { 4 }, { 2 } },
+    { VA_FOURCC_UYVY, 1, { 4 }, { 2 } },
+    { VA_FOURCC_RGBX, 1, { 8 }, { 2 } },
+    { VA_FOURCC_RGBA, 1, { 8 }, { 2 } },
+    { VA_FOURCC_BGRX, 1, { 8 }, { 2 } },
+    { VA_FOURCC_BGRA, 1, { 8 }, { 2 } },
+};
+
+static void getPlaneWidth(uint32_t w, uint32_t plane, const uint32_t multiple[3], uint32_t width[3])
```

Comment:
Think about this before, but I do not find a better name, Do you have suggestion?